### ISSUE TITLE
Separate minimal/base project

### ISSUE THREAD

**Issue by axl303** [Source2ZE/CS2Fixes#242]
Hello guys, 
Is it possible to make a separate project under Source2ZE profile, but to be just a base for CS2 plugin, like just a simple admin system, simple commands and map vote system or something like that? 
You got my point I think so?
I think that currently people are downloading the source and removing a lot of unnecessary things to adapt it for their idea - like 5vs5/practice/public/jump/bhop/kz or whatever mod.
If there is just a base repository with minimal things a server mod/plugin to work with the minimal things would be much, much more beautiful.
Something like SourceMod or AMXX plugins?

**Comment by xLeviNx** [comment]
Have you used it yet? There's commands that literally enable and disable stuff. Fits the needs of the majority. But also, this is more in tune to be developed for zombie escape.

**Comment by axl303** [comment]
Yes, I am currently using it ONLY for learning purposes and some private 5vs5 and practice servers between friends.
Since a lot of stuff is being changed I can't really keep up with the new stuff/functions chaning and everytime I download the whole source code when much things/commits are done and removing the unnecessary.
So a an idea came to my mind to ask if it is possible to be created something like my above comment. Would be much easier for people to develop from there, and this one repository to be for the zombie/zombie escape etc..

**Comment by xLeviNx** [comment]
With cs2 constantly changing, wouldn't recommend that the base project targeted for ze be changed for that. Ideally you could just enable what you want and disable what you don't want. There are cvars that disable almost every function that is made by cs2fixes. 

Basically saying it's a waste of time and an inconvenience

**Comment by axl303** [comment]
> With cs2 constantly changing, wouldn't recommend that the base project targeted for ze be changed for that.

In issue title "Separate" <- / targeted only for simple things and users to be able to develop how they want from there, not to remove things that are meant for zombie etc.

> Ideally you could just enable what you want and disable what you don't want. There are cvars that disable almost every function that is made by cs2fixes.

I totally agree with you, but I will stick with my first comment/question.

**Comment by Vauff** [comment]
The idea of splitting the project up into different plugin components has become more appealing to me lately. However, there is still a fair bit of cross-communication needed between our code, and designing around this/building out API will take a fair bit of work/coordination. There are some other considerations too. Nobody else really seemed interested in pursuing this yet.

**Comment by axl303** [comment]
> The idea of splitting the project up into different plugin components has become more appealing to me lately.

I think that one repository with this base things "may be like Starter pack :D", the base like commands, admin commands, map system, the events (maybe and a bit more) will be good, so people (mainly developers or newbie ones like me) to learn and to build their own ideas from there. Just a simple metamod plugin (C++) if you didn't understand me or if I can't explain it good enough.

> different plugin components

I also think that this is good, but may be time, work wasting

> designing around this/building out API will take a fair bit of work/coordination

API is time taking and wasting it for this I think? I mean wasting work and time
